actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -617,7 +621,7 @@ def current_controller
         def use_recall_for(key)
           if @recall[key] && (!@options.key?(key) || @options[key] == @recall[key])
             if !named_route_exists? || segment_keys.include?(key)
-              @options[key] = @recall.delete(key)
+              @options[key] = @recall[key]
```

Comment:
See https://github.com/rails/rails/issues/24438

what was overlooked is a few mutations to ```@recall```
https://github.com/rails/rails/blob/eba0041c2132e46bf9e040b31059e020e590d2ae/actionpack/lib/action_dispatch/routing/route_set.rb#L577
https://github.com/rails/rails/blob/eba0041c2132e46bf9e040b31059e020e590d2ae/actionpack/lib/action_dispatch/routing/route_set.rb#L636

this is now going to mutate the recall passed in by callers.